framework/include/interfaces/DiffusionHDGInterface.h

```diff
@@ -0,0 +1,476 @@
+
+  // Material properties
+  const MaterialProperty<Real> & _diff;
+
+  // Number of dofs on elem
+  std::size_t _vector_n_dofs;
+  std::size_t _scalar_n_dofs;
+  std::size_t _lm_n_dofs;
+
+  /// Local sizes of the systems
+  std::size_t _primal_size, _lm_size;
+
+  /// Our stabilization coefficient
+  static constexpr Real _tau = 1;
+};
+
+template <typename DiffusionHDG>
+void
+DiffusionHDGInterface::resizeData(DiffusionHDG & obj)
+{
+  obj._vector_n_dofs = obj._qu_dof_indices.size();
+  obj._scalar_n_dofs = obj._u_dof_indices.size();
+  obj._lm_n_dofs = obj._lm_u_dof_indices.size();
+
+  libmesh_assert_equal_to(obj._vector_n_dofs, obj._vector_phi.size());
+  libmesh_assert_equal_to(obj._scalar_n_dofs, obj._scalar_phi.size());
+
+  obj._primal_size = obj._vector_n_dofs + obj._scalar_n_dofs;
+  obj._lm_size = obj._lm_n_dofs;
+
+  // prepare our matrix/vector data structures
+  obj._PrimalMat.setZero(obj._primal_size, obj._primal_size);
+  obj._PrimalVec.setZero(obj._primal_size);
+  obj._LMMat.setZero(obj._lm_size, obj._lm_size);
+  obj._LMVec.setZero(obj._lm_size);
+  obj._PrimalLM.setZero(obj._primal_size, obj._lm_size);
+  obj._LMPrimal.setZero(obj._lm_size, obj._primal_size);
+}
+
+template <typename DiffusionHDG>
+void
+DiffusionHDGInterface::vectorVolumeResidual(DiffusionHDG & obj,
+                                            const unsigned int i_offset,
+                                            const MooseArray<Gradient> & vector_sol,
+                                            const MooseArray<Number> & scalar_sol)
+{
+  for (const auto qp : make_range(obj._qrule->n_points()))
+    for (const auto i : make_range(obj._vector_n_dofs))
+    {
+      // Vector equation dependence on vector dofs
+      obj._PrimalVec(i_offset + i) += obj._JxW[qp] * (obj._vector_phi[i][qp] * vector_sol[qp]);
+
+      // Vector equation dependence on scalar dofs
+      obj._PrimalVec(i_offset + i) += obj._JxW[qp] * (obj._div_vector_phi[i][qp] * scalar_sol[qp]);
+    }
+}
+
+template <typename DiffusionHDG>
+void
+DiffusionHDGInterface::vectorVolumeJacobian(DiffusionHDG & obj,
+                                            const unsigned int i_offset,
+                                            const unsigned int vector_j_offset,
+                                            const unsigned int scalar_j_offset)
+{
+  for (const auto qp : make_range(obj._qrule->n_points()))
+    for (const auto i : make_range(obj._vector_n_dofs))
+    {
+      // Vector equation dependence on vector dofs
+      for (const auto j : make_range(obj._vector_n_dofs))
+        obj._PrimalMat(i_offset + i, vector_j_offset + j) +=
+            obj._JxW[qp] * (obj._vector_phi[i][qp] * obj._vector_phi[j][qp]);
+
+      // Vector equation dependence on scalar dofs
+      for (const auto j : make_range(obj._scalar_n_dofs))
+        obj._PrimalMat(i_offset + i, scalar_j_offset + j) +=
+            obj._JxW[qp] * (obj._div_vector_phi[i][qp] * obj._scalar_phi[j][qp]);
```

Comment:
is there an equation in the doc we can tie to this?

Reply:
yes, adding some